plugins/spoiler-alert/spec/system/composer_spoiler_spec.rb

```diff
@@ -0,0 +1,226 @@
+# frozen_string_literal: true
+
+describe "Composer - ProseMirror editor - Spoiler extension", type: :system do
+  fab!(:current_user) do
+    Fabricate(
+      :user,
+      refresh_auto_groups: true,
+      composition_mode: UserOption.composition_mode_types[:rich],
+    )
+  end
+
+  let(:cdp) { PageObjects::CDP.new }
+  let(:composer) { PageObjects::Components::Composer.new }
+  let(:rich) { composer.rich_editor }
+
+  def click_spoiler_button
+    find(".toolbar-menu__options-trigger").click
+    find("[title='#{I18n.t("js.spoiler.title")}']").click
+  end
+
+  before { sign_in(current_user) }
+
+  def open_composer
+    page.visit "/new-topic"
+    expect(composer).to be_opened
+    composer.focus
+  end
+
+  it "creates inline spoiler when cursor is at end of line" do
+    open_composer
+
+    composer.type_content("This is a test ")
+    click_spoiler_button
+
+    expect(rich).to have_css("span.spoiled", text: I18n.t("js.composer.spoiler_text"))
+
+    composer.type_content("spoiler")
+
+    expect(rich).to have_css("span.spoiled", text: "spoiler")
+
+    composer.toggle_rich_editor
+
+    expect(composer).to have_value("This is a test [spoiler]spoiler[/spoiler] ")
+  end
+
+  it "creates block spoiler when cursor is at start of line" do
+    open_composer
+
+    click_spoiler_button
+
+    expect(rich).to have_css("div.spoiled", text: I18n.t("js.composer.spoiler_text"))
+
+    composer.type_content("spoiler")
+    composer.toggle_rich_editor
+
+    expect(composer).to have_value("[spoiler]\nspoiler\n\n[/spoiler]\n\n")
+  end
+
+  it "creates block spoiler with input rule when cursor is at start of line" do
+    open_composer
+    composer.type_content("[spoiler]")
+    composer.type_content("inside")
+    composer.toggle_rich_editor
+
+    expect(composer).to have_value("[spoiler]\ninside\n\n[/spoiler]\n\n")
+  end
+
+  it "wraps selected text in inline spoiler when selection is within same paragraph" do
+    open_composer
+
+    composer.type_content("This is **secret** text here")
+    find("strong", text: "secret").double_click
+    click_spoiler_button
+
+    expect(rich).to have_css("span.spoiled strong", text: "secret")
+    expect(rich).to have_content("This is secret text here")
+
+    composer.toggle_rich_editor
+    expect(composer).to have_value("This is [spoiler]**secret**[/spoiler] text here")
+  end
+
+  it "wraps selected text in block spoiler when selection spans multiple paragraphs" do
+    open_composer
+
+    composer.type_content("First paragraph")
+    composer.send_keys(:enter)
+    composer.type_content("Second paragraph")
+    composer.select_all
+    click_spoiler_button
+
+    expect(rich).to have_css("div.spoiled")
```

Comment:
do we need this one? if we don't have this, the next one is not going to pass anyways